### ISSUE TITLE
`enforce-canonical-classes` removes unrelated conflicting classes

### ISSUE THREAD

**Issue by codeflorist** [schoero/eslint-plugin-better-tailwindcss#304]
### Description

When a non-canonical class is used together with (unrelated) conflicting classes (e.g. `flex grid aspect-[4/3]`, a fix will automatically remove the conflicting classes - resulting in just e.g. `aspect-4/3`.

This is also visible via the warning:
`The classes: "flex, grid, aspect-[4/3]" can be simplified to "aspect-4/3"`

### Flavor

HTML

### Code Input

```tsx
<h1 class="flex grid aspect-[4/3]">
    Hello world!
  </h1>
```

### Expected Behavior

```tsx
`aspect-[4/3]` should get fixed to `aspect-4/3`, but `flex grid` should remain, be not automatically fixed and throw an error.
```

### Actual Behavior

```tsx
`flex grid aspect-[4/3]` gets automatically fixed into `aspect-4/3`.
```

### ESLint Log

```shell
9:14  warning  The classes: "flex, grid, aspect-[4/3]" can be simplified to "aspect-4/3"                 better-tailwindcss/enforce-canonical-classes 
  9:14  error    Conflicting class detected: "flex" and "grid" apply the same CSS properties: ""display""  better-tailwindcss/no-conflicting-classes    
  9:19  warning  The classes: "flex, grid, aspect-[4/3]" can be simplified to "aspect-4/3"                 better-tailwindcss/enforce-canonical-classes 
  9:19  error    Conflicting class detected: "grid" and "flex" apply the same CSS properties: ""display""  better-tailwindcss/no-conflicting-classes    
  9:24  warning  The classes: "flex, grid, aspect-[4/3]" can be simplified to "aspect-4/3"                 better-tailwindcss/enforce-canonical-classes 

✖ 5 problems (2 errors, 3 warnings)
  0 errors and 3 warnings potentially fixable with the `--fix` option.
```

### ESLint Config

```typescript
Basic setup according to https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/main/docs/parsers/html.md
```

### Versions

- ESLint: 9.39.2
- Parser: `"@html-eslint/parser": "^0.53.0"`
- Plugin: 4.0.1
- Node: v22.20.0
